recipes/xkbcommon/all/conanfile.py

```diff
@@ -56,23 +56,23 @@ def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
-    def validate(self):
-        if self.settings.os not in ["Linux", "FreeBSD"]:
-            raise ConanInvalidConfiguration("This library is only compatible with Linux or FreeBSD")
-
     def requirements(self):
         self.requires("xorg/system")
         if self.options.get_safe("xkbregistry"):
             self.requires("libxml2/2.9.12")
-        if self.options.get_safe("with_wayland", False):
-            self.requires("wayland/1.19.0")
-            self.requires("wayland-protocols/1.21")  # FIXME: This should be a build-requires
+        if self.options.get_safe("with_wayland"):
+            self.requires("wayland/1.20.0")
+            self.requires("wayland-protocols/1.24")  # FIXME: This should be a build-requires
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            raise ConanInvalidConfiguration("This library is only compatible with Linux or FreeBSD")
 
     def build_requirements(self):
-        self.build_requires("meson/0.59.1")
-        self.build_requires("bison/3.7.1")
-        if self.options.get_safe("with_wayland", False):
-            self.build_requires("wayland/1.19.0")
+        self.build_requires("meson/0.60.2")
+        self.build_requires("bison/3.7.6")
+        if hasattr(self, "settings_build") and self.options.get_safe("wayland"):
```

Comment:
why the addition of `hasattr(self, "settings_build")` ?

Reply:
it's explained in the last commit

Reply:
usual issue when a downstream recipe overrides wayland with 1 profile

Reply:
ok, thanks